types/ember__polyfills/index.d.ts

```diff
@@ -13,6 +13,7 @@ import { Mix, Mix3, Mix4 } from './types';
 export function assign<T extends object, U extends object>(target: T, source: U): Mix<T, U>;
 export function assign<T extends object, U extends object, V extends object>(target: T, source1: U, source2: V): Mix3<T, U, V>;
 export function assign<T extends object, U extends object, V extends object, W extends object>(target: T, source1: U, source2: V, source3: W): Mix4<T, U, V, W>;
+export function assign(target: object, ...sources: object[]): any;
```

Comment:
`any` includes the ability for the return to be undefined, null, boolean, etc... You should probably return `object` here.